.github/workflows/doc_links_checker.yml

```diff
@@ -0,0 +1,28 @@
+name: Doc Link Checker
```

Comment:
Does this need a license at the top?

Reply:
We need a license header specified [here](https://github.com/opensearch-project/.github/issues/21). The header for new created file is 
```
# Copyright OpenSearch Contributors
# SPDX-License-Identifier: Apache-2.0
```
However, it could not pass the line check and will trigger an error:
```
error  File must start with a license header  @osd/eslint/require-license-header
```
I think we need to clarify what the header should be for new created file. No issue for modified file.

Reply:
@ananzh  So we have a license header check somewhere? Do you know how to trigger it? My initial thought these license header will only be treat as header comments